src/rules/flake8_django/rules.rs

```diff
@@ -0,0 +1,225 @@
+use crate::ast::types::Range;
+use crate::define_violation;
+use crate::registry::Diagnostic;
+use crate::violation::Violation;
+use ruff_macros::derive_message_formats;
+use rustpython_ast::Constant::Bool;
+use rustpython_ast::ExprKind::{Attribute, Call, Constant, Name};
+use rustpython_ast::StmtKind::{Assign, ClassDef, FunctionDef};
+use rustpython_ast::{Expr, Located, Stmt};
+
+define_violation!(
+    pub struct ModelDunderStr;
+);
+impl Violation for ModelDunderStr {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Model does not define __str__ method")
+    }
+}
+
+impl ModelDunderStr {
+    pub fn check_dunder_str(
```

Comment:
The convention in the codebase is to write these as top-level functions with the same name as the violation, rather than associated methods on the violation, e.g.:

```rust
pub fn model_dunder_str(bases: &[Expr], ...) -> Option<Diagnostic> {
  ...
}
```